src/parser/mod.rs

```diff
@@ -313,42 +315,87 @@ fn at_rule2(input: Span) -> IResult<Span, Item> {
     }
 }
 
+fn forward2(input: Span) -> IResult<Span, Item> {
+    let (mut input, path) =
+        terminated(any_sass_string, opt_spacelike)(input)?;
+    let mut found_as = None;
+    let mut expose = Expose::All;
+    let mut found_with = None;
+    while let (rest, Some(arg)) = opt(terminated(name, opt_spacelike))(input)?
+    {
+        input = match arg.as_ref() {
+            "as" if found_as.is_none() => {
+                let (i, a) = terminated(as_arg, opt_spacelike)(rest)?;
+                found_as = Some(a);
+                i
+            }
+            "hide" if expose == Expose::All => {
+                let (i, (funs, vars)) = exposed_names(rest)?;
+                expose = Expose::Hide(funs, vars);
+                i
+            }
+            "show" if expose == Expose::All => {
+                let (i, (funs, vars)) = exposed_names(rest)?;
+                expose = Expose::Show(funs, vars);
+                i
+            }
+            "with" if found_with.is_none() => {
+                let (i, w) = with_arg(rest)?;
+                found_with = Some(w);
+                i
+            }
+            other => panic!("Unimplemented @forward arg: {}", other),
```

Comment:
This needs to fail the parser (return an error) properly, rather than just panic.